runtime/lib/ttnn/operations/creation/ones.cpp

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ones.h"
```

Comment:
I feel like we should just use full op for this. Under the hood ttnn implements it as a full op with fill value 1. This way we can reuse the code of full op for different ops that target specific fill values (zeros, ones, twos etc.).

Not sure what the best way to do this on the compiler/flatbuffer side is though. We could just always use full with specific values, or we could add a ones op in the ttnn dialect that then lower it to full with specific values when translating to flatbuffer. So the flatbuffer will only have a schema for full.

Reply:
Actually I think a better way to do this is to use the current full op implementation as an API. We can add zeros, ones etc. to full.h/full.cpp, and just execute it with specific fill values. This way we get all the implementation for free when we want to add new variants/specifications for full op, and we can ensure consistency across all of them.

Currently we have:
```
void run(const ::tt::target::ttnn::FullOp *op, ProgramContext &context) {
  ProgramTensorPool &tensorPool = context.getTensorPool();
  FullTensorConfig config(op);
  ::ttnn::Tensor out;
  const ::tt::target::DeviceRef *deviceRef =
      !utils::inSystemMemory(op->out()) ? op->device() : nullptr;

  if (config.numShards == 1) {
    out = createFullOnSingleDevice(context, config, deviceRef);
  } else if (config.numShards > 1) {
    out = createFullOnMultiDevice(context, config, deviceRef);
  } else {
    LOG_FATAL("Unsupported num shards");
  }
  tensorPool.insert_or_assign(op->out()->global_id(), out);
}
``` 
We can update `FullTensorConfig` to explicitly take in all the parameters instead of taking in the op and deriving the parameters, and it'd be up to the `run` function of each op to derive these parameters. 
Since `FullOp` currently doesn't have layout, dtype etc. it would need to derive from the output tensor descriptor, but `Ones` can just get it from the op, and explicitly use `1` as the fill Value.


Then every op would just have the following code in common:
```
  if (config.numShards == 1) {
    out = createFullOnSingleDevice(context, config, deviceRef);
  } else if (config.numShards > 1) {
    out = createFullOnMultiDevice(context, config, deviceRef);
  } else {
    LOG_FATAL("Unsupported num shards");
  }
  tensorPool.insert_or_assign(op->out()->global_id(), out);
 ```
 Or we could even just wrap all this code in an API like `executeFull` or something

Reply:
Jackson OOO right now and I don't have strong opinion, so will approve this for now to unblock, let's circle back and discuss this situation in the future like you said.